### DISCUSSION TITLE
Error: No valid AXIS lines found by GriddedData

### DISCUSSION THREAD

**Discussion by tstepan333** [idaholab/moose#19026] [Q&A Modules: Solid mechanics] [answered]
I'm trying to compile one of the examples for [PiecewiseMultilinear](https://mooseframework.inl.gov/source/functions/PiecewiseMulticonstant.html) function. I copied .i and .txt files to my directory. However, when I try to compile the file, I'm getting the error 'No valid AXIS lines found by GriddedData'. Could you help me to resolve this issue?

Files:
https://github.com/idaholab/moose/blob/next/test/tests/functions/piecewise_multilinear/twoDa.i
https://github.com/idaholab/moose/blob/next/test/tests/functions/piecewise_multilinear/twoD_const.txt

**Comment by WilkAndy** [comment, marked as answer]
I'm not sure what you've done exactly.  The MOOSE input file `twoDa.i` does not use the data file `twoD_const.txt`.  Instead, it uses `twoD1.txt`, because of the block

```
  [./bilinear1_fcn]
    type = PiecewiseMultilinear
    data_file = twoD1.txt
  [../]
```

So, i suspect that you've copied the files, but then modified stuff.  Check your modifications.

andy

**Reply by tstepan333** [reply]
I downloaded files again to my directory and run them w/o modifications. However, I'm still getting the same error. All the files I copy from /moose I store on my computer and compile them with ./giraffe-opt -i /mnt/c/Users/Tatiana/Desktop/MOOSE/test/twoD.i

**Reply by WilkAndy** [reply]
Did you also copy `twoD1.txt`, since that is the file that is used by `twoDa.i`.

**Reply by tstepan333** [reply]
Yes, if there is no twoD1.txt file the error is 'Error opening file '/mnt/c/Users/Tatiana/Desktop/MOOSE/test/twoD1.txt' from GriddedData.'

**Reply by GiudGiud** [reply]
This really should not be failing. This is a tested input. i just checked with a fresh MOOSE and it runs fine. 
Did you change anything wrt to these systems in Giraffe?

**Reply by tstepan333** [reply]
No, I didn't do anything. Will try to update moose

**Reply by GiudGiud** [reply]
This was added in 2018, I doubt this was ever broken since then. But always a good idea
